File: closest_to_zero/closest.cpp
#include<iostream>
#include<algorithm>
#include<vector>

class Solution {
public:
    int findClosestNumber(std::vector<int>& nums) {
        int closest=nums[0];
        
        for(auto num:nums){
             if(abs(num)<abs(closest)) closest=num;
        }
        
        if(closest<0&&std::find(nums.begin(),nums.end(),abs(closest))!=nums.end()){
            return abs(closest);
        }else{
            return closest;
        }
        
    }
};

File: closest_to_zero/closest.py
class Solution(object):
    def findClosestNumber(self, nums):
        closest=nums[0]
        for num in nums:
            if abs(num)<abs(closest):
                closest=num
        
        if(closest<0 and abs(closest) in nums):
            return abs(closest)
        return closest
        